lacework/resource_lacework_team_member.go

```diff
@@ -0,0 +1,557 @@
+						},
+						"user": {
+							Type:          schema.TypeBool,
+							Optional:      true,
+							Default:       false,
+							ConflictsWith: []string{"organization.0.administrator"},
+							Description:   "Whether the team member is an org level user",
+						},
+						"admin_accounts": {
+							// We are not using Set because we need to use DiffSuppressFunc: https://github.com/hashicorp/terraform-plugin-sdk/issues/160
+							Type:             schema.TypeList,
+							Optional:         true,
+							Description:      "List of accounts the team member is an admin",
+							ConflictsWith:    []string{"organization.0.user", "organization.0.administrator"},
+							DiffSuppressFunc: diffCaseInsensitive,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+								StateFunc: func(val interface{}) string {
+									return strings.ToUpper(strings.TrimSpace(val.(string)))
+								},
+							},
+						},
+						"user_accounts": {
+							// We are not using Set because we need to use DiffSuppressFunc: https://github.com/hashicorp/terraform-plugin-sdk/issues/160
+							Type:             schema.TypeList,
+							Optional:         true,
+							Description:      "List of accounts the team member is a user",
+							ConflictsWith:    []string{"organization.0.user", "organization.0.administrator"},
+							DiffSuppressFunc: diffCaseInsensitive,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+								StateFunc: func(val interface{}) string {
+									return strings.ToUpper(strings.TrimSpace(val.(string)))
+								},
+							},
+						},
+					},
+				},
+			},
+			"guid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceLaceworkTeamMemberCreate(d *schema.ResourceData, meta interface{}) error {
+	lacework := meta.(*api.Client)
+
+	if lacework.OrgAccess() {
+		return laceworkTeamMemberCreateOrg(d, meta)
+	}
+
+	return laceworkTeamMemberCreate(d, meta)
+}
+
+func laceworkTeamMemberCreateOrg(d *schema.ResourceData, meta interface{}) error {
+	lacework := meta.(*api.Client)
+
+	tmOrg := api.NewTeamMemberOrg(d.Get("email").(string),
+		api.TeamMemberProps{
+			AccountAdmin: d.Get("administrator").(bool),
+			Company:      d.Get("company").(string),
+			FirstName:    d.Get("first_name").(string),
+			LastName:     d.Get("last_name").(string),
+		})
+
+	if !d.Get("enabled").(bool) {
+		tmOrg.UserEnabled = 0
+	}
+
+	tmOrg.OrgAdmin = d.Get("organization.0.administrator").(bool)
+	tmOrg.OrgUser = d.Get("organization.0.user").(bool)
+	tmOrg.AdminRoleAccounts = castAndUpperStringSlice(d, "organization.0.admin_accounts")
+	tmOrg.UserRoleAccounts = castAndUpperStringSlice(d, "organization.0.user_accounts")
```

Comment:
Is there a reason to not set these fields inside NewTeamMemberOrg ?

Reply:
Not a good reason, no. We should be able to move this with https://lacework.atlassian.net/browse/ALLY-798